chapter01/1.3 - URLify/urlify.js

```diff
@@ -0,0 +1,24 @@
+var urlify = function(str, length) {
+  // have a pointer to check from start to end
+  var strArr = str.split('');
+  var pointer = 0;
+  while (pointer < str.length) {
+    if (strArr[pointer] === ' ') {
+      // *** needs more work here, a little wierd
+      // not handling trailing spaces properly
+      for (var i = str.length - 1; i > pointer + 3; i--) {
+        strArr[i] = str[i - 2];
+      }
+      strArr[pointer] = '%';
+      strArr[pointer+1] = '2';
```

Comment:
Hey @dawsonbotsford, for this question in particular, if we used replace or strip, then we wouldn't have to deal with iterating through the string character by character, as well as right-shifting the characters. In the context of an interview, if an interviewer wanted to see if we knew how to do string manipulation and the character level, using JavaScript methods wouldn't be useful for the interviewer to understand our programming abilities. So I think it really depends.

I generally approach questions according to its requirements, because I am of the view that while interviewers do their best to be language agnostic, they usually have their programming paradigms primed towards the language which they are most comfortable with, so if that language is not JavaScript, we need to take that into consideration so as to communicate effectively with the interviewer.

That being said, force-fitting JavaScript traits into questions designed for other languages does prove awkward sometimes. For this question, I was attempting to leave trailing whitespace in the string, which was unnatural for JavaScript. Also, JavaScript strings are also immutable, which led to the need to split up the string into an array of characters. I have also encountered awkwardness when doing bit manipulation questions.